pyhealth/trainer.py

```diff
@@ -271,9 +272,16 @@ def inference(self, dataloader) -> Dict[str, float]:
                 loss_all.append(loss.item())
                 y_true_all.append(y_true)
                 y_prob_all.append(y_prob)
+                if additional_outputs is not None:
+                    for key in additional_outputs.keys():
+                        additional_outputs[key].append(output[key].cpu().numpy())
```

Comment:
Should we add `if key in output` in the middle?

Reply:
I'm leaning to not but am OK with either. I think `additional_outputs` is used only for users/developers who know what they are doing, and if we don't trigger errors it could be hard to debug.